API.md

```diff
@@ -1542,14 +1542,26 @@ const schema = Joi.object({
 });
 ```
 
-#### `string.regex(pattern, [name])`
+#### `string.regex(pattern, [name | config])`
 
 Defines a regular expression rule where:
 - `pattern` - a regular expression object the string value must match against.
 - `name` - optional name for patterns (useful with multiple patterns). Defaults to 'required'.
+- `config` - an optional configuration object with the following supported properties:
+  - `name` - optional pattern name. Defaults to `required`.
```

Comment:
Same as the original, I have absolutely no idea where that `Defaults to required` came from, it isn't :)